celery/app/task.py

```diff
@@ -882,8 +882,13 @@ def replace(self, sig):
             )
 
         if self.request.chain:
+            options = sig.options.copy()
```

Comment:
This appears to fix `test_chain_replaced_with_a_chain_and_an_error_callback()` AFAICT
```suggestion
            options = sig.options.copy()
            sig.freeze(self.request.id)
```